scala-thoughts.md

```diff
@@ -0,0 +1,51 @@
+#Scala Thoughts#
+
+###What's easy to do in Scala?  What's not?###
+
+Functional programming seems relatively easy in Scala.  In this respect, the
+language had many similarities to other functional languages like Racket and
+Haskell.  I didn't explicitly have to say "return", as is true of other functional
+languages.  The use of higher-order functions was similar to my experiences with
+Racket as well.  Scala also does support pattern matching like Haskell, but it
+seemed slightly more cumbersome.  This made it very easy to, for example, return
+different results based on different inputs, as we did in the `matchOnInputType`
+function.  It was also very easy to declare variables and use type inference, at
+least in the scope of this assignment.  Also, the collections seemed to have
+extensive and intuitives APIs, supported by thorough documentation online.  This
+made it very easy to take advantage of the built-in functionality of lists.
+
+###What are your favorite choices that the designers of Scala made? Why?###
+
+There were a few design choices that I very much appreciated.  I appreciated
+the different ways you could use pattern matching.  In match/case statements,
+it was possible to include a line like `case x : Int if x > 0`, so that we
```

Comment:
That's a great point! For me, it was weird at first because pattern matching with a case statement felt foreign. You're right though that it's a very concise way to express the thought and would be harder or at least less concise in most other languages.